framework/endpoints/cart_api.py

```diff
@@ -0,0 +1,106 @@
+            body:  data for deleting items from shopping cart with required fields
+                  {
+                    "shoppingCartItemIds": ["CartItemId"]
+
+                  }
+            expected_status_code: expected http status code from response
+            token: JWT token for authorization of request
+        """
+
+        headers = self.headers
+        headers["Authorization"] = f"Bearer {token}"
+        path = self.url + "/items"
+        response = requests.delete(url=path, headers=headers, data=json.dumps(body))
+        assert_that(response.status_code, is_(expected_status_code),
+                    reason=f"Expected status code {expected_status_code}, found: {response.status_code}")
+        log_request(response)
+
+        return response
+
+    def update_quantity_product(self, token: str, item_id: str, quantity: int,
+                                expected_status_code: int = 200) -> Response:
+        """Updating product's quantity
+
+                Args:
+                    expected_status_code: expected http status code from response
+                    token: JWT token for authorization of request
+                    item_id: product id to update
+                    quantity: quantity of product for updating
+
+                """
+
+        data = {
+            "shoppingCartItemId": item_id,
+            "productQuantityChange": quantity
+        }
+        headers = self.headers
+        headers["Authorization"] = f"Bearer {token}"
+        path = self.url + "/items"
+        response = requests.patch(url=path, data=json.dumps(data), headers=headers)
+        assert_that(response.status_code, is_(expected_status_code),
+                    reason=f"Expected status code {expected_status_code}, found: {response.status_code}")
+        log_request(response)
+
+        return response
+
+    def add_new_item_to_cart(self, token: str, items: list, expected_status_code: int = 200) -> Response:
+        """Adding multiple products to cart and verifying if they are in the response
+
+        Args:
+            items: A list of items where each item is a dictionary containing 'productId' and 'productQuantity'
+            expected_status_code (int): Expected HTTP status code from response
+            token (str): JWT token for authorization of request
+
+        Example:
+            items = [
+                {"productId": "123ffg-333-jjj78", "productQuantity": 2},
+                {"productId": "6788gg-uh8-hajj6", "productQuantity": 3}
+            ]
+        """
+
+        headers = self.headers
+        headers["Authorization"] = f"Bearer {token}"
+        path = self.url + "/items"
+        body = {"items": items}
+        response = requests.post(url=path, data=json.dumps(body), headers=headers)
+        assert_that(response.status_code, is_(expected_status_code),
+                    reason=f"Expected status code {expected_status_code}, found: {response.status_code}")
+        log_request(response)
```

Comment:
My goal was to keep the response status code inside the function, so if someone forgets to include a status code check in the test, it will be in the method. Another point for this method is that you can save yourself time by not writing down this statement every time you use this method. What do you think?

Reply:
Fixed. Added exist method for assertion http status code to the api method